lib/logstash/inputs/s3.rb

```diff
@@ -134,7 +134,8 @@ def list_new_files
           @logger.debug('S3 Input: Object Zero Length', :key => log.key)
         elsif !sincedb.newer?(log.last_modified)
           @logger.debug('S3 Input: Object Not Modified', :key => log.key)
-        elsif log.storage_class.start_with?('GLACIER')
+        elsif log.storage_class.start_with?('GLACIER') &&
```

Comment:
do you think the restore also concern `DEEP_ARCHIVE` in any ways?
there was a quick attempt to completely skip those: 
https://github.com/logstash-plugins/logstash-input-s3/pull/177/files#diff-7d2cf6058313d9dde77cbdb3b92d3bffR137